### ISSUE TITLE
resolution failure when dependency locks and  kotlin-dsl plugin are present

### ISSUE THREAD

**Issue by rpalcolea** [gradle/gradle#10697]
Given the following build file:

```
plugins {
    `kotlin-dsl`
}

dependencyLocking {
    lockAllConfigurations()
}

repositories {
    gradlePluginPortal()
}

java {
    targetCompatibility = JavaVersion.VERSION_1_8
    sourceCompatibility = JavaVersion.VERSION_1_8
}
```

when running `./gradlew dependencies --write-locks`, the `compileClasspath` configuration has the following:

```
compileClasspath - Compile classpath for compilation 'main' (target  (jvm)).
Persisted dependency lock state for configuration ':compileClasspath'
+--- org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.3.41
|    \--- org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.3.41
|         \--- org.jetbrains.kotlin:kotlin-stdlib:1.3.41
|              \--- org.jetbrains:annotations:13.0
\--- org.jetbrains.kotlin:kotlin-reflect:1.3.41
     \--- org.jetbrains.kotlin:kotlin-stdlib:1.3.41 (*)
``` 

This results in this lock file:

```
# This is a Gradle generated file for dependency locking.
# Manual edits can break the build and are not advised.
# This file is expected to be part of source control.
org.jetbrains.kotlin:kotlin-reflect:1.3.41
org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.3.41
org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.3.41
org.jetbrains.kotlin:kotlin-stdlib:1.3.41
org.jetbrains:annotations:13.0
``` 

This looks fine, however when running `./gradlew build`, `compileKotlin` fails with the following:

```
Execution failed for task ':compileKotlin'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Resolved 'org.jetbrains.kotlin:kotlin-stdlib-common:1.3.41' which is not part of the dependency lock state
``` 

Running `./gradlew dependencies` this time will show a different graph:

```

compileClasspath - Compile classpath for compilation 'main' (target  (jvm)).
+--- org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.3.41
|    +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.41
|    |    +--- org.jetbrains.kotlin:kotlin-stdlib-common:1.3.41
|    |    \--- org.jetbrains:annotations:13.0
|    \--- org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.3.41
|         \--- org.jetbrains.kotlin:kotlin-stdlib:1.3.41 (*)
+--- org.jetbrains.kotlin:kotlin-reflect:1.3.41
|    \--- org.jetbrains.kotlin:kotlin-stdlib:1.3.41 (*)
+--- org.jetbrains.kotlin:kotlin-reflect:{strictly 1.3.41} -> 1.3.41 (c)
+--- org.jetbrains.kotlin:kotlin-stdlib-jdk8:{strictly 1.3.41} -> 1.3.41 (c)
+--- org.jetbrains.kotlin:kotlin-stdlib:{strictly 1.3.41} -> 1.3.41 (c)
+--- org.jetbrains.kotlin:kotlin-stdlib-jdk7:{strictly 1.3.41} -> 1.3.41 (c)
+--- org.jetbrains:annotations:{strictly 13.0} -> 13.0 (c)
\--- org.jetbrains.kotlin:kotlin-stdlib-common:1.3.41 FAILED
``` 

`kotlin-stdlib-common` was introduced during compilation.

The current workaround is to add the dependency manually to the lock files `org.jetbrains.kotlin:kotlin-stdlib-common:<version>`.

Wondering if this is related to: https://github.com/gradle/gradle/blob/master/subprojects/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/base/KotlinDslBasePlugin.kt#L48 and https://github.com/gradle/gradle/blob/master/subprojects/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/base/KotlinDslBasePlugin.kt#L58

**Comment by ljacomet** [comment]
Issue for 6.0 fix: #10745

**Comment by rpalcolea** [comment]
awesome! thanks for the quick fix @ljacomet

**Comment by eskatos** [comment]
Fixed for Gradle 5.6.3 in https://github.com/gradle/gradle/pull/10691
